File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        //brute force
//         if (s.length() != t.length()) return false;
//         sort(s.begin(), s.end());
//         sort(t.begin(), t.end());
//         return s == t;   
        
        
        // if (s.length() != t.length()) return false;
        // vector<int> freq(26, 0);
        // for (int i=0; i<s.length(); i++){
        //     freq[s[i] - 'a']++;
        //     freq[t[i] - 'a']--;
        // }
        // for (const int c: freq)
        //     if (c != 0) return false;
        // return true;
        
        //Two freq approach
        if (s.size() != t.size())
            return false;
        vector<int> freq1(26, '0');
        vector<int> freq2(26, '0');
        for (int i=0; i<s.size(); i++){
            freq1[s[i] - 'a']++;
            freq2[t[i] - 'a']++;
        }
        if (freq1.size() != freq2.size())
            return false;
        for (int i=0; i<freq1.size(); i++) {
            if (freq1[i] != freq2[i])
                return false;
        }
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        freq_s = {}
        freq_t = {}
        for i in s:
            freq_s[i] = freq_s.get(i, 0) + 1
        for i in t:
            freq_t[i] = freq_t.get(i, 0) + 1
        return freq_s == freq_t
        
        
        